x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -241,6 +241,11 @@ synchronized AsyncSearchResponse toAsyncSearchResponse(AsyncSearchTask task, lon
      * @return response representing the status of async search
      */
     synchronized AsyncStatusResponse toStatusResponse(String asyncExecutionId, long startTime, long expirationTime) {
+        SearchResponse.Clusters clustersInStatus = null;
+        if (clusters != null && clusters.getTotal() > 0) {
```

Comment:
Can we have a clusters instance that has total == 0 in practice?

Reply:
Yes, TransportSearchAction uses Clusters.EMPTY for [executeLocalSearch](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java#L313). 

That is why for searches against local clusters only we don't display the `_clusters` object. It's content-free for that scenario.

The CCS pathway (even with a local cluster) doesn't use that pathway in TransportSearchAction. Instead, it sets up an informative Clusters object.

Reply:
good point. I forgot.